CMakeLists.txt

```diff
@@ -93,36 +57,46 @@ message(STATUS "List of arduino boards ${ARDUINO_DIR}")
 ################################################################################
 # Animal related options.
 ################################################################################
-if( (NOT ANIMAL_NAME) OR (NOT SESSION_TYPE) OR (NOT SESSION_NUM) )
-    message( FATAL_ERROR 
-        "Please pass ANIMAL_NAME, SESSION_NUM and SESSION_TYPE to cmake \n"
-        "\te.g cmake -DANIMAL_NAME=k3 -DSESSION_NUM=1 -DSESSION_TYPE=S .. \n"
+if((NOT ANIMAL_NAME) OR (NOT SESSION_TYPE) OR (NOT SESSION_NUM) OR (NOT PROTO_CODE))
+    message(FATAL_ERROR 
+        "Set ANIMAL_NAME, SESSION_NUM, SESSION_TYPE, and PROTO_CODE to cmake; e.g. \n"
+        "\tcmake -DANIMAL_NAME=k3 -DSESSION_NUM=1 -DSESSION_TYPE=S -DPROTO_CODE=All1 ..\n"
```

Comment:
The `SESSION_TYPE` has been changed to character instead of numbers. @ananthamurthy Please note this. Currently `S` (sound), `L` (light), and `M` (mixed) session types are allowed. Talk to others if this needs to be expanded.

Reply:
Hey, I didn't get this part. Currently we have many session types. How will s,l,m suffice?